### ISSUE TITLE
Testing google backend on travis

### ISSUE THREAD

**Issue by hobarrera** [pimutils/vdirsyncer#560]
I'd like to see if it's possible to run CI with google's API.

My main idea is:

* Have a `client_id` and `client_secret` as encrypted travis variables.
* Keep the refresh token in some place that lets us store stuff online*. I really wish we could have *real* long-lived tokens, but oh, well.
* Run tests like we do with other servers (well, actually, requires setting a token and stuff, so a bit of extra work involved).

I'm willing to work on this, since I *have* to use google's caldav auth for work stuff recently.

* Firebase DB? I recall Mozilla having a similar database (where you could store and retrieve arbitrary JSON), but I can't seem to find it.

**Comment by untitaker** [comment]
Last time I looked into this [there appeared to be a way to create tokens programmatically](https://developers.google.com/identity/protocols/OAuth2ServiceAccount), i.e. without that very interactive auth dance.

**Comment by untitaker** [comment]
To be clear @hobarrera are you still interested in working on this?

**Comment by hobarrera** [comment]
Yes, sorry, I've just been busier that expected lately, but definitely!
